src/Components/test/testassets/BasicTestApp/RouterTest/NavLinkIgnoreQueryAndFragmentString.razor

```diff
@@ -0,0 +1,27 @@
+@using Microsoft.AspNetCore.Components.Routing;
+@inherits NavLink
+
+@{
+    base.BuildRenderTree(__builder);
+}
+
+@code{
+
+    protected override bool ShouldMatch(string currentUriAbsolute)
+    {
+        if (base.ShouldMatch(currentUriAbsolute))
+        {
+            return true;
+        }
+
+        var originalMatch = base.Match;
+        base.Match = NavLinkMatch.All;
+        string uriWithoutQueryAndFragment = GetUriIgnoreQueryAndFragment(currentUriAbsolute);
+        var shouldMatch = base.ShouldMatch(uriWithoutQueryAndFragment);
+        base.Match = originalMatch;
+        return shouldMatch;
+    }
+
+    private string GetUriIgnoreQueryAndFragment(string uri) =>
+        new Uri(uri).GetLeftPart(UriPartial.Path);
+}
```

Comment:
Instead of doing this dance, just write a plain C# class, it's much less code and doesn't require tricks like
```
@{
    base.BuildRenderTree(__builder);
}
```